app/common/renderer/utils/commands-tab.js

```diff
@@ -0,0 +1,229 @@
+import _ from 'lodash';
+
+import {APPIUM_TO_WD_COMMANDS, COMMANDS_WITH_MISMATCHED_PARAMS} from '../constants/commands.js';
+
+/**
+ * Try to detect if the input value should be a boolean/number/array/object,
+ * and if so, convert it to that
+ *
+ * @param {string} value the value of a command parameter
+ * @returns the value converted to the type it matches best
+ */
+export function adjustParamValueType(value) {
+  if (value === '') {
+    return null;
+  } else if (Number(value).toString() === value) {
+    return Number(value);
+  } else if (['true', 'false'].includes(value)) {
+    return value === 'true';
+  } else {
+    try {
+      return JSON.parse(value);
+    } catch {
+      return value;
+    }
+  }
+}
+
+/**
+ * Filter the array of method key-value pairs to only include methods matching the search query.
+ *
+ * @param {[string, Object][]} methodPairs array of [methodName, methodDetails] pairs
+ * @param {string} searchQuery user-provided search query
+ * @returns filtered array of [methodName, methodDetails] pairs
+ */
+export function filterMethodPairs(methodPairs, searchQuery) {
+  return _.filter(methodPairs, ([methodName]) => {
+    if (!searchQuery) {
+      return true;
+    }
+    return methodName.toLowerCase().includes(searchQuery.toLowerCase());
+  });
+}
+
+/**
+ * Check if a value is an empty object or array ({} or [])
+ *
+ * @param {*} value any value (object, array, primitive)
+ * @returns whether the item is an empty object/array or not
+ */
+const isEmptyObject = (value) => _.isObjectLike(value) && _.isEmpty(value);
+
+/**
+ * Recursively remove key/value pairs (or array entries) whose values are empty objects or arrays.
+ * If this causes the parent object/array to become empty,
+ * this parent will be removed from its own parent as well.
+ *
+ * @param {*} value any value (object, array, primitive)
+ * @returns the value with empty entries removed
+ */
+export function deepFilterEmpty(value) {
+  if (_.isArray(value)) {
+    // Recurse into each array element, then remove empty entries
+    const mapped = value.map((v) => deepFilterEmpty(v));
```

Comment:
may be simplified to

```
return value.map(deepFilterEmpty).filter(_.negate(isEmptyObject))
```

Reply:
Just in case you are into functional programming, check the concept of lodash's flow: https://www.geeksforgeeks.org/javascript/lodash-_-flow-method/